mautic/mautic
# Relative parameters in the date string segment filter are shown in different languages

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


Segment date filters do not work properly after switching to another language in the configuration.  

E.g., the relative date `today` turns into a date in local format after switching from English to German.
![image](https://github.com/mautic/mautic/assets/96085911/c956694b-835e-4759-be01-f5ed530a8ab9)
![image](https://github.com/mautic/mautic/assets/96085911/8448294e-8c73-43ed-b017-6898ae6ed5d5)

E.g., the relative date `heute` value  turns into 1970-01-01 00:00` after switching from German to English.
![image](https://github.com/mautic/mautic/assets/96085911/4cc2ed71-fc7a-4a6c-8937-d71d5b6cb1f9)
![image](https://github.com/mautic/mautic/assets/96085911/4d5f4891-5959-45cb-acd7-18138e061512)

This PR suggests storing relative dates in English in DB and converting them to the current language in UI.

This PR is similar to PR [10541](https://github.com/mautic/mautic/pull/10541), but it allows to display relative date strings in different languages.
#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Set English as a current language.
3. Create a new segment, add a filter with date field and set a relative value, e.g. `Date Last Active` `equals` `today`.
4. Save the segment.
5. Switch to another language, e.g. German.
6. Check the segment. Relative value translates to current language, e.g. `heute`.
7. Create a new segment, add a filter with date field and set a relative value, e.g. `Letzte Aktivität Datum` `gleich` `heute`.
8. Save the segment.
9. Switch to another language, e.g. English.
10. Check the segment. Relative value translates to current language, e.g. `today`.
11. Run command `mautic:segment:update`. It works well.
